packages/server/lib/controllers/oauth.controller.ts

```diff
@@ -652,6 +653,15 @@ class OAuthController {
     public async oauthCallback(req: Request, res: Response, _: NextFunction) {
         const { state } = req.query;
 
+        const installation_id = req.query['installation_id'] as string | undefined;
+        const action = req.query['setup_action'] as string;
+
+        if (!state && installation_id && action) {
```

Comment:
We redirect here likely back to Github because if there is no state then the process was initiated from Github and so we have no notion of the user within the context of Nango